compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/FileSystemRepository.java

```diff
@@ -172,9 +173,54 @@ public FileSystemRepository(Environment environment, Path cacheDirectory, String
         } catch (IOException e) {
             throw new RuntimeException("Error while accessing Distribution cache: " + e.getMessage());
         }
+
+        List<Path> incompatibleVersions = new ArrayList<>();
+
+        if (!versions.isEmpty()) {
+            for (Path ver : versions) {
+                Path pkgJsonPath = getPackagePath(org.value(), name.value(), ver.getFileName().toString())
```

Comment:
`ver.getFileName()` can be `null`. We should wrap it with `Optional.of`

Reply:
made use of Optional to resolve spotbugs issue